src/triage/component/catwalk/predictors.py

```diff
@@ -342,28 +260,54 @@ def predict(self, model_id, matrix_store, misc_db_parameters, train_matrix_colum
                 session.close()
 
         model = self.load_model(model_id)
-        logger.spam(f"Loaded model {model_id}")
         if not model:
-            raise ModelNotFoundError(f"Model # {model_id} not found")
+            raise ValueError(f"Model id {model_id} not found")
+        logger.spam(f"Loaded model {model_id}")
 
         # Labels are popped from matrix (i.e. they are removed and returned)
         labels = matrix_store.labels
 
-        predictions_proba = model.predict_proba(
+        predictions = model.predict_proba(
             matrix_store.matrix_with_sorted_columns(train_matrix_columns)
-        )
+        )[:, 1]  # Returning only the scores for the label == 1
+
+
         logger.debug(
             f"Generated predictions for model {model_id} on {matrix_store.matrix_type.string_name} matrix {matrix_store.uuid}"
         )
         if self.save_predictions:
+            df = pd.DataFrame(data=None, columns=None, index=matrix_store.index)
+            df['label_value'] = matrix_store.labels
+            df['score'] = predictions
+
+
+            logger.spam(f"Sorting predictions for model {model_id} using {self.rank_order}")
+
+            if self.rank_order == 'best':
+                df.sort_values(by=["score", "label_value"], inplace=True, ascending=[False,False], na_position='last')
+            elif self.rank_order == 'worst':
+                df.sort_values(by=["score", "label_value"], inplace=True, ascending=[False,True], na_position='first')
+            elif self.rank_order == 'random':
+                df.sample(frac=1).sort_values(by='score', inplace=True, ascending=False)
+            else:
+                raise ValueError(f"Rank order specified in condiguration file not recognized: {self.rank_order} ")
+
+            df['rank_abs_no_ties'] = df['score'].rank(ascending=False, method='first')
+            df['rank_abs_with_ties'] = df['score'].rank(ascending=False, method='min')
+            df['rank_pct_no_ties'] = df['rank_abs_no_ties'].rank(ascending=True, method='dense', pct=True)
+            df['rank_pct_with_ties'] = df['score'].rank(ascending=False, method='dense', pct=True)
```

Comment:
might be worth including a comment here with some of the logic. For instance, why is `rank_abs_with_ties` using `min` while `rank_pct_with_ties` using `dense`? Also, I assume `rank_pct_no_ties` starts from `rank_abs_no_ties` to ensure these have the same sort order, but might be good to call that out.

Reply:
Done

Reply:
![](https://media1.giphy.com/media/83QtfwKWdmSEo/200.gif)

Reply:
thanks!